### PR TITLE
Cast alloca and temporary variables to expected address space

### PR DESCRIPTION
**Pull request by yxsamliu**

### CODE REVIEW

File: lib/CodeGen/CodeGenFunction.h [hunk 1]
```diff
@@ -1950,12 +1950,16 @@ class CodeGenFunction : public CodeGenTypeCache {
   ///
   /// That is, this is exactly equivalent to CreateMemTemp, but calling
   /// ConvertType instead of ConvertTypeForMem.
-  Address CreateIRTemp(QualType T, const Twine &Name = "tmp");
+  Address CreateIRTemp(QualType T, const Twine &Name = "tmp",
+      bool KeepSAddrSpace = true);
```
Comment (whchung) [inline lib/CodeGen/CodeGenFunction.h]: @yxsamliu could you clarify about the name "KeepSAddrSpace"? looks like a typo to me.

Comment (yxsamliu) [reply lib/CodeGen/CodeGenFunction.h]: It is explained in the comment of CreateMemTemp.

Comment (whchung) [reply lib/CodeGen/CodeGenFunction.h]: I mean Keep`S`AddrSpace

**Comment by yxsamliu** [comment]
sorry. typo. fixed.

**Review by whchung — APPROVED** [review]
for future upstream sync purpose, this PR is derived from: https://reviews.llvm.org/D32248

**Pull request merged**